src/bundle.ts

```diff
@@ -7,88 +7,141 @@ import { pathToPointer } from './pathToPointer';
 import { pointerToPath } from './pointerToPath';
 import { traverse } from './traverse';
 
-export const BUNDLE_ROOT = '__bundled__';
-export const ERRORS_ROOT = '__errors__';
-
-export const bundleTarget = <T = unknown>({ document, path }: { document: T; path: string }, cur?: unknown) =>
-  _bundle(cloneDeep(document), path, { [path]: true }, cur);
-
-const _bundle = (
-  document: unknown,
-  path: string,
-  stack: Dictionary<boolean>,
+export const BUNDLE_ROOT = '#/__bundled__';
+export const ERRORS_ROOT = '#/__errors__';
+
+export const bundleTarget = <T = unknown>(
+  {
+    document,
+    path,
+    bundleRoot = BUNDLE_ROOT,
+    errorsRoot = ERRORS_ROOT,
+  }: { document: T; path: string; bundleRoot?: string; errorsRoot?: string },
   cur?: unknown,
-  bundledRefInventory: any = {},
-  bundledObj: any = {},
-  errorsObj: any = {},
 ) => {
-  const objectToBundle = get(document, pointerToPath(path));
-
-  traverse(cur ? cur : objectToBundle, ({ parent }) => {
-    if (hasRef(parent) && isLocalRef(parent.$ref)) {
-      const $ref = parent.$ref;
-      if (errorsObj[$ref]) return;
+  if (`${path}/`.startsWith(`${bundleRoot}/`) || `${path}/`.startsWith(`${errorsRoot}/`)) {
+    throw new Error(`Roots do not make any sense`);
+  }
 
-      if (bundledRefInventory[$ref]) {
-        parent.$ref = bundledRefInventory[$ref];
+  return bundle(cloneDeep(document), bundleRoot, errorsRoot)(path, { [path]: true }, cur);
+};
 
-        // no need to continue, this $ref has already been bundled
-        return;
-      }
+const bundle = (document: unknown, bundleRoot: string, errorsRoot: string) => {
+  const bundleRootPath = pointerToPath(bundleRoot);
+  const errorsRootPath = pointerToPath(errorsRoot);
+  const scopedBundledObj = get(document, bundleRootPath);
+
+  const takenKeys = new Set<string | number>(
+    typeof scopedBundledObj === 'object' && scopedBundledObj !== null ? [...Object.keys(scopedBundledObj)] : [],
+  );
+
+  const _bundle = (
+    path: string,
+    stack: Dictionary<boolean>,
+    cur?: unknown,
+    bundledRefInventory: any = {},
+    bundledObj: any = {},
+    errorsObj: any = {},
+  ) => {
+    const $refTarget = pointerToPath(path);
+    // const $bundleKey = $refTarget[$refTarget.length - 1];
```

Comment:
We can probably remove it? or does it provide you any value Marc?

Reply:
yeah can remove

Reply:
```suggestion
```